docs/sphinx/developer/hpctoolkit.rst

```diff
@@ -0,0 +1,192 @@
+   $ print(filtered_gf_rev.tree(metric_column = "time (inc)")) #printing the new filtered subtree by inclusive time metric
+ 
+   $ filtered_gf.drop_index_levels() #As-is, the tree will include info for ranks - if that isn't needed, this function drops that info
+   $ filtered_gf.dataframe #this provides a spreadsheet of the data that is populating the graphframe (what the tree shows)
+   $ filtered_gf.dataframe.iloc[0] #gives the first entry of the spreadsheet, here that is the root node of the filtered tree
+   $ filtered_gf.dataframe.iloc[0,0] #gives the first part of the first entry of the spreadsheet (here, it's the inclusive time) 
+
+   $ gf3 = filtered_gf - filtered_gf_rev #Stores the diff between two (comparable) trees in gf3
+   $ print(gf3.tree()) #prints the diff tree
+   $ gf3.dataframe #outputs the spreadsheet of data that populates the diff tree
+
+
+This example was set up to analyze the performance of the ``no-op_stress_test.cpp`` benchmark file
+from the Umpire repo. It compares the performance from one part of the program (i.e., the part that 
+measure the performance when doing deallocations in the "same order" as they were allocated) versus 
+another part of the same program (i.e., the part that measures the performance when doing deallocations
+in the "reverse order" as they were allocated). 
+
+In Hatchet, these two parts show up as subtrees within the entire call path tree of my example program.
+Therefore, I can compare one subtree to another in terms of performance (in my case, I compared in terms
+of inclusive time).
+
+Analyzing results
+------------------
+
+After opening up a database folder in HPCViewer or analyzing the call paths in Hatchet, we can compare the performance
+(or whatever measurement we are looking at) of different parts of the program against other parts and try to find 
+performance trends. In Umpire's use case, we plan to use Hatchet as part of our CI to find out if integrating a new
+commit into our repository increases the performance by a certain threshold or more. If so, our CI test will fail. Our
+process looks something like:
+
+* Grab the example program's database from the develop branch
+* Grab the example program's database from a different branch I want to compare against
+* Create a graphframe for each database
+* Create a filtered graphframe for each that focuses on the specific part of the program I want to measure against
+* Compare the inclusive time for each filtered graphframe (or whatever metric I want to analyze)
+* If the metric (e.g., inclusive time) of the new branch's filtered graphframe is more than ``threshold`` more than that of develop's, then fail the test!
+
+This is just one example of how analyzing results from HPCToolKit and Hatchet can be used to improve performance of Umpire.
+There are many other ways to do this - once we have more interesting use cases, this page will be updated!
+
+.. note::
```

Comment:
This should be removed.

Reply:
Done

Reply:
I meant this whole note should be removed. I don't think linking to JIRA from readthedocs is necessary or appropriate